src/ImageSharp/Formats/Tiff/Writers/TiffPaletteWriter{TPixel}.cs

```diff
@@ -55,23 +55,38 @@ internal sealed class TiffPaletteWriter<TPixel> : TiffBaseColorWriter<TPixel>
         /// <inheritdoc />
         protected override void EncodeStrip(int y, int height, TiffBaseCompressor compressor)
         {
-            Span<byte> pixels = GetStripPixels(((IPixelSource)this.quantizedImage).PixelBuffer, y, height);
+            Span<byte> indexedPixels = GetStripPixels(((IPixelSource)this.quantizedImage).PixelBuffer, y, height);
             if (this.BitsPerPixel == 4)
             {
-                using IMemoryOwner<byte> rows4bitBuffer = this.MemoryAllocator.Allocate<byte>(pixels.Length / 2);
+                int width = this.Image.Width;
+                int excess = (width % 2) * height;
+                int rows4BitBufferLength = indexedPixels.Length + excess;
```

Comment:
it seems that here you can:
`int rows4BitBufferLength = (indexedPixels.Length / 2) + excess;`

Reply:
yeah your right, this was a bit too much